wgu-opensource/openedx-mfe-learning
# feat: Configurable Header Logo Link and Footer Links

# Overview

This PR adds an optional link from the header logo to a configurable dashboard URL.
It also adds a configuration option to override the default links in the footer (Privacy Policy, Terms of Service, and Honor Code)

## Test Steps
* Set following configuration options in `.env.private`:
```
DISABLE_APP_FOOTER=false
DISABLE_DESKTOP_HEADER=false
DISABLE_HEADER_LOGO=false
FOOTER_LINK_MAP='[{"label": "Test Link", "url": "https://wgu.edu/", "id": 1}]'
ENABLE_DASHBOARD_RETURN_LINK=true
DASHBOARD_BASE_URL='https://academy.test'
```
* Check that the only footer URL is now "Test Link" and links to the specified URL
* Check that the header links to https://academy.test/course/overview/<course-id>

## Checklist

⚠️ **Please make sure to fill this checklist before asking for reviews.**

- [x] Code is correctly formatted and linted
- [x] Unit tests are updated and are passing
- [x] PR title follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary):
      i.e. `feat(CUR-###): Feature title`
- [x] PR Title aligns to Semantic-Release [supported prefixes](https://www.conventionalcommits.org/en/v1.0.0/#examples) (NOTE: prefixes are case sensitive, keep lower-case):

```diff
 feat() - Feature (0.X.0)
 fix() - Patch (0.0.X)
 docs() - Patch (0.0.X), will only scope to README changes
 refactor() - Patch (0.0.X)
 revert() - Patch (0.0.X)
 style() - Patch (0.0.X)
```

- [x] Check for unused files
- [x] Check work before asking for reviewers
- [x] Fix any linting errors
- [x] SECURITY: No secrets where committed to the repo
- [x] COMPLIANCE: Committed code is not proprietary and adheres to Open Source licensing
